src/components/care/WaterButton.tsx

```diff
@@ -20,6 +20,10 @@ export function WaterButton() {
   const handleSelect = (itemId: string) => {
     actions.updateState((currentState) => {
       const result = waterPet(currentState, itemId);
+      if (!result.success) {
+        // Use setTimeout to ensure alert shows after state update
+        setTimeout(() => alert(result.message), 0);
```

Comment:
Native `window.alert` will halt JS thread and prevent tick processing from progressing normally. It also provides a worse UI experience for the user. Use a dialog instead.